src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -172,11 +172,6 @@ public PinValue Read(int pinNumber)
                 throw new InvalidOperationException("Can not read from a pin that is not open.");
             }
 
-            if (_driver.GetPinMode(logicalPinNumber) == PinMode.Output)
```

Comment:
Will this work for all drivers? I believe the reason why we h ad this in the first place is because even though SysFS case would work, other drivers might not. We should test all drivers before going ahead and doing this change.

Reply:
Also, we have a test that is guarding the behavior that we will throw if attempting to read from an output pin, so if we decide to take this, then we would have to go and modify/remove that test.

Reply:
I also think we should leverage that. I haven't seen a case where reading from an output pin would be a problem.

Reply:
If there may be a driver not supporting this, I feel that it would make sense to throw this exception only from there instead of denying it for everyone.